### PR TITLE
Fixes #35621 - Make power status a selectable column

### PR DESCRIPTION
**Pull request by ofedoren**

### CODE REVIEW

File: app/registries/foreman/settings/general.rb [hunk 1]
```diff
@@ -35,11 +35,6 @@
       description: N_("Text to be shown in the login-page footer. Keyword $VERSION is replaced by current version."),
       default: N_("Version") + " $VERSION",
       full_name: N_('Login page footer text'))
-    setting('host_power_status',
-      type: :boolean,
-      description: N_("Show power status on host index page. This feature calls to compute resource providers which may lead to decreased performance on host listing page."),
-      default: true,
-      full_name: N_('Show host power status'))
```
Comment (adamruzicka) [inline app/registries/foreman/settings/general.rb:42]: Should we also clean the stored value from db, if there is any?

**Review by adamruzicka — COMMENTED** [review]
Do we have any sane way of emulating the setting? Just to avoid the "I had it turned off and now I'm seeing the column again" kind of scenario

**Comment by ofedoren** [comment]
Thanks, @adamruzicka, updated with migration to delete the setting from DB.



> Do we have any sane way of emulating the setting? Just to avoid the "I had it turned off and now I'm seeing the column again" kind of scenario

I'm afraid I don't have any good idea about that...

**Comment by adamruzicka** [comment]
Needs a rebase

**Comment by ofedoren** [comment]
Thanks, @adamruzicka, updated.

**Comment by ares** [comment]
This would be a great 3.5 improvement (the major feature is already there, this would make it even more powerful). If it is safe, I'd vote for merging even during the stabilization.

**Comment by ares** [comment]
I have tested this and from the user perspective it works great. Based on the code, we now render the power status component even when the column is hidden, therefore we always create ~20 requests to power status API (for each row). Each such request takes ~20ms on my setup. It's not too bad but something to consider before the merging, I'd personally be OK with it at least for now, this is the default behavior of Foreman today anyway. We may optimize this later.

**Comment by adamruzicka** [comment]
Agreed, we should get rid of the hidden columns completely once all the plugins had time to move from deface to proper selectable columns, but we can live with it for the time being

**Review by adamruzicka — APPROVED** [review]

**Comment by adamruzicka** [comment]
Thank you @ofedoren & @ares !

**Comment by ekohl** [comment]
One regression in this patch is that previously it wouldn't even request the power status if the column was hidden but now every host in the host overview still sends an API call to show the power status.

**Comment by ofedoren** [comment]
Well, that's a regression just because of backwards compatibility. As of me I'd remove compatibility lines. @ShimShtein, what do you think? Asking since you've introduced https://github.com/theforeman/foreman/commit/fdb9a92ed832a78f35eb599c39b1f4bb8ccd4f4c.

**Comment by ShimShtein** [comment]
The regression is caused by this patch because it removed the `power_column_visible?` method for the backwards compatible columns.
Anyway, I can agree with @ofedoren, its value as a marker for deface customizations is not great to begin with, since the column is not always there.

**Pull request merged**